### PR TITLE
Add free disk space log on docker install error

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Add free disk space info method.
- Add info log about available space on docker image install error.

### CODE REVIEW

File: supervisor/host/info.py [hunk 1]
```diff
@@ -66,6 +67,11 @@ async def get_dmesg(self) -> bytes:
 
         return stdout
 
+    def get_disk_free_space(self, path: str = "/data") -> float:
```
Comment (MartinHjelmare) [inline supervisor/host/info.py:70]: Is `/data` the path that is used for pulling docker images?

Comment (MartinHjelmare) [reply supervisor/host/info.py:70]: Maybe use this property instead?
https://github.com/home-assistant/supervisor/blob/e1ce19547e0a0d056d645436aa291e86fbabbe9b/supervisor/config.py#L135-L138

File: supervisor/docker/interface.py [hunk 2]
```diff
@@ -99,6 +99,8 @@ def _install(
                 docker_image.tag(image, tag="latest")
         except docker.errors.APIError as err:
             _LOGGER.error("Can't install %s:%s -> %s.", image, tag, err)
+            free_space = self.coresys.host.info.get_disk_free_space()
```
Comment (frenck) [inline supervisor/docker/interface.py:102]: We could check `err.response.status_code == 404:` here and add some additional message in the log.

e.g., `This error is often caused by not having enough disk space available.`

**Comment by timmo001** [comment]
I had this exact problem the other night. Caused all sorts of problems on my server. Thanks for this! 👍

File: supervisor/docker/interface.py [hunk 3]
```diff
@@ -99,6 +99,13 @@ def _install(
                 docker_image.tag(image, tag="latest")
         except docker.errors.APIError as err:
             _LOGGER.error("Can't install %s:%s -> %s.", image, tag, err)
+            if err.status_code == 404:
+                free_space = self.coresys.host.info.get_disk_free_space()
```
Comment (pvizeli) [inline supervisor/docker/interface.py:103]:
```suggestion
                free_space = self.sys_host.info.get_disk_free_space()
```

File: supervisor/host/info.py [hunk 4]
```diff
@@ -66,6 +68,13 @@ async def get_dmesg(self) -> bytes:
 
         return stdout
 
+    def get_disk_free_space(self, path: Optional[Union[str, Path]] = None) -> float:
```
Comment (pvizeli) [inline supervisor/host/info.py:71]: Let's make an which use fix the path_supervisor
```
@property
free_space(self) -> float:
```

And put the `get_disk_free_space` into hardware as helper with path.

**Comment by pvizeli** [comment]
perfect 🐬

**Pull request merged**